routing/payment_lifecycle.go

```diff
@@ -332,6 +350,13 @@ lifecycle:
 		return htlc.Settle.Preimage, &htlc.Route, nil
 	}
 
+	// At this point the payment should have a failure reason and therefore
+	// this should never happen.
+	if failure == nil {
```

Comment:
Have you encountered this case before?

Reply:
yeah i had that concern too - tho I think it's not possible to have this being `nil`. A better way is to [Defining errors out of existence](https://web.stanford.edu/~ouster/cgi-bin/cs190-spring15/lecture.php?topic=errorHandling) so we don't need to even worry about it, maybe a new PR to change the pointer to be a concrete struct.